src/it/resources/com/google/checkstyle/test/chapter2filebasic/rule231filetab/InputFormattedWhitespaceCharacters.java

```diff
@@ -0,0 +1,39 @@
+package com.google.checkstyle.test.chapter2filebasic.rule231filetab;
+
+final class InputFormattedWhitespaceCharacters {
+  // Long line ----------------------------------------------------------------
+  // Contains a tab ->	<- // violation 'Line contains a tab character.'
+  // Contains trailing whitespace ->
+
+  /**
+   * Some javadoc.
+   *
+   * @param badFormat1 bad format
+   * @param badFormat2 bad format
+   * @param badFormat3 bad format
+   * @return hack
+   * @throws Exception abc
+   */
+  int test1(int badFormat1, int badFormat2, final int badFormat3) throws Exception {
+    return 0;
+  }
+
+  // A very, very long line that is OK because it matches the regexp "^.*is OK.*regexp.*$"
+  // long line that has a tab ->	<- and would be OK if tab counted as 1 char
+  // violation above 'Line contains a tab character.'
+
+  // tabs that count as one char because of their position ->	<-   ->	<-
+  // violation above 'Line contains a tab character.'
```

Comment:
formatter wasn't able to remove tabs from such places

Reply:
is there an opened issue in the formatter?

Reply:
Please create issue and we will move forward

Reply:
issue is created https://github.com/google/google-java-format/issues/1149